### PR TITLE
create API & types for notifications

### PR DESCRIPTION
**Pull request by EvgeniyaYakunina**
### **Description:** 
This pull request introduces a new notificationsApi with endpoints for managing notifications within the application. The API includes functionality to retrieve notifications, mark them as read, and delete specific notifications by ID.

### **Key Changes:**
deleteNotification: Mutation to delete a notification using its unique ID. Invalidates the Notifications tag to ensure the client has the latest notification data after deletion.
getAllNotifications: Query to fetch all notifications, with optional cursor-based pagination. Provides the Notifications tag to automatically update when changes are made.
markNotificationAsRead: Mutation to mark specified notifications as read, accepting an array of notification IDs. Invalidates the Notifications tag to refresh state.
Motivation and Context:
These endpoints provide essential functionality for the notifications feature, supporting user engagement by allowing users to view, manage, and delete notifications as needed.

### **Usage:**
**Hooks added:**
useDeleteNotificationMutation for deleting a notification
useGetAllNotificationsQuery for retrieving notifications
useMarkNotificationAsReadMutation for marking notifications as read

### CODE REVIEW

**Review by NoName-ForTeam — APPROVED** [review]

**Review by Lelik7777 — APPROVED** [review]

File: src/features/profile/api/notifications/notificationsApi.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      query: ({ cursor }) => GET_ALL_NOTIFICATIONS(cursor),
+    }),
+    /** Marks a list of notifications as read.
+     * @param {number[]} params.ids - Array of notification IDs to mark as read.
+     */
+    markNotificationAsRead: builder.mutation<void, { ids: number[] }>({
+      invalidatesTags: ['Notifications'],
+      query: params => ({
+        body: params,
+        method: POST,
```
Comment (AramZakaryan) [inline src/features/profile/api/notifications/notificationsApi.ts]: метод PUT, правильно понимаю?

Comment (EvgeniyaYakunina) [reply src/features/profile/api/notifications/notificationsApi.ts]: Да, все верно) Моя ошибка. Благодарю за внимательность!

**Comment by EvgeniyaYakunina** [comment]
run build production

**Comment by EvgeniyaYakunina** [comment]
run build production

**Pull request merged**